device-identifier/shbdeviceidentifier/app.py

```diff
@@ -1,26 +1,235 @@
+@click.option("--verbose", "-v", help="Enable verbose output.", **flag_default_options)
+@click.option("--version", "version_flag", help="Show current app version.", **flag_default_options)
+@pass_ctx
+def app(ctx, debug, silent, verbose, version_flag):
+    """
+    SmartHomeBuddy's device identifier.
+    """
+    if silent:
+        # TODO: Implement silent mode
+        ...
+
+    if verbose:
+        ctx.verbose = True
+        logger.level("INFO")
+    if debug:
+        logger.level("DEBUG")
+
+    if version_flag:
+        click.echo(f"Version: {ctx.version}")
+        sys.exit(0)
+
+    figlet = Figlet(font='smslant', justify='left')  # choose btw: small, stampatello, smslant
+    click.echo(figlet.renderText('SmartHomeBuddy'))
+
+    # run_rpc_server()
+    ctx.server_running = True
+    logger.success("RPC server started.")
+
+
+# ---------------------------------------------------------------------------- #
+#                                 Commands                                     #
+# ---------------------------------------------------------------------------- #
+
+@app.command("start")
+@pass_ctx
+@logger_wraps()
+def start(ctx):
+    """
+    Starts the RPC server.
+    """
+    if not ctx.server_task:
+        run_rpc_server()
+        ctx.server_running = True
+
+
+@app.command("stop")
+@pass_ctx
+@logger_wraps()
+def stop(ctx):
+    """
+    Stops the RPC server.
+    """
+    ctx.server_task.cancel()
+    logger.success("RPC server stopped.")
+
+
+@app.command("collect")
+@click.option("-f", "--file_path", type=click.Path(), required=False, default="")
+@pass_ctx
+@logger_wraps()
+def collect(ctx, file_path):
+    """
+    Collects all the data from an interface.
+    """
+    file_path = check_file_path(ctx, file_path)
+    ...
+
 
+@app.command("read")
+@click.option("-f", "--file_path", type=click.Path(), required=False, default="")
+@click.option('--file-type', '-T', help='File type to read.', required=False,
+              type=click.Choice(['pcap', 'pcapng'], case_sensitive=False), default='pcap')
+@pass_ctx
+@logger_wraps()
+def read(ctx, file_path, file_type):
+    """
+    Reads all the data from a capture file.
+    """
+    file_path = check_file_path(ctx, file_path)
+    ...
 
-def _get_args() -> argparse.Namespace:
-    parser = argparse.ArgumentParser(description="SmartHomeBuddie's device identifier.")
-    parser.add_argument(
-        "--collect", dest="collect_mode", action="store_true", help="Collect network packages."
-    )
-    parser.set_defaults(collect_mode=False)
-    return parser.parse_args()
 
+@app.command("identify")
+@click.option("-f", "--file_path", type=click.Path(), required=False, default="")
+@click.option("-o", "--out", type=click.Path(), required=False, default="influx")  # TODO: refactor
+@pass_ctx
+@logger_wraps()
+def identify(ctx, file_path, out):
+    """
+    Identifies a device.
+    """
+    file_path = check_file_path(ctx, file_path)
+    ...
 
-def app() -> None:
```

Comment:
Do we not need `app()`anymore?

Reply:
Has been replaced by line 130 in the new file. Still there.